Extensions/DialogueTree/dialoguetools.js

```diff
@@ -654,8 +676,8 @@ gdjs.dialogueTree.compareVariable = function(key, value) {
  * @param {string} value The value you want to set
  */
 gdjs.dialogueTree.setVariable = function(key, value) {
-  if (this.dialogueIsRunning && this.runner.variables.data) {
-    this.runner.variables.data[key] = value;
+  if (this.runner.variables) {
+    this.runner.variables.set(key, JSON.parse(value));
```

Comment:
Why do you want to parse `value`?

Reply:
To preserve the type bondagejs expects them to be. If something is set with yarn syntax, bondage converts it automatically to a type. Now imagine we want to overwrite that same bondagejs variable from the event sheet. We need to do the same in order to preserve its type

Reply:
to see what I mean, open yarn and set a bunch of different variables with the types boolean,int, float and string.

Then console.log(this.runner.variables) - this is where bondagejs stores/reads them.

Now run the game and go through the yarn node where the yarn syntax is setting them. You will find that the variables object gets populated with them in different types.They are not all just strings